faiss/IndexFlat.cpp

```diff
@@ -129,6 +129,10 @@ struct FlatL2Dis : DistanceComputer {
         return fvec_L2sqr(b + j * d, b + i * d, d);
     }
 
+    inline const float *data(idx_t i) override {
```

Comment:
please have it return a void* so that it is clear for the caller that the data may be compressed (eg. with a scalar quantizer).